crates/router/src/core/payments.rs

```diff
@@ -330,6 +339,47 @@ impl PaymentRedirectFlow for PaymentRedirectCompleteAuthorize {
     fn get_payment_action(&self) -> services::PaymentAction {
         services::PaymentAction::CompleteAuthorize
     }
+
+    fn generate_response(
+        &self,
+        payments_response: api_models::payments::PaymentsResponse,
+        merchant_account: storage_models::merchant_account::MerchantAccount,
+        payment_id: String,
+        connector: String,
+    ) -> RouterResult<api::RedirectionResponse> {
+        // There might be multiple redirections needed for some flows
+        // If the status is requires customer action, then send the startpay url again
+        // The redirection data must have been provided and updated by the connector
+        match payments_response.status {
+            api_models::enums::IntentStatus::RequiresCustomerAction => {
+                let startpay_url = payments_response
+                    .next_action
+                    .and_then(|next_action| next_action.redirect_to_url)
+                    .ok_or(errors::ApiErrorResponse::InternalServerError)
+                    .into_report()
+                    .attach_printable(
+                        "did not receive redirect to url when status is requires customer action",
+                    )?;
+                Ok(api::RedirectionResponse {
+                    return_url: String::new(),
```

Comment:
@Narayanbhat166 Can we pass return_url as empty string?

Reply:
Yes return_url_with_query_params is the one being sent in response